app/src/main/java/nl/tudelft/cs4160/trustchain_android/main/OverviewConnectionsActivity.java

```diff
@@ -346,6 +351,9 @@ private void startSendThread() {
 
             while(true) {
                 try {
+                    // update connection type and internal ip address
+                    network.updateConnectionType((ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE));
```

Comment:
I got a crash here because updateConnectionType sets a TextView from a secondary thread

Reply:
Strange that it crashes on that. Because it happens on the main (UI) thread, so I don't know what the problem is. I could change it so it's just getConnectionType, because this construct is kind off weird anyway. However then the update still happens on another thread.

Reply:
Oh wait, what it means is that it doesn't get set from the main thread. So there should be a runOnUIThread lambda there.